src/nvim/os/env.c

```diff
@@ -456,6 +456,15 @@ char *vim_getenv(const char *name)
     if (p_hf != NULL && vim_strchr(p_hf, '$') == NULL) {
       vim_path = (char *)p_hf;
     }
+
+#ifdef WIN32
+    char exe_name[MAXPATHL];
+    if (vim_path == NULL) {
+      GetModuleFileName(NULL, exe_name, MAXPATHL);
+      vim_path = exe_name;
```

Comment:
The Vim version has `#ifdef USE_EXE_NAME` blocks which are defined for OS X and Windows, but not unix. It does things like this also:

    pend = remove_tail(p, pend, (char_u *)"src");

The logic should  be wrapped in a check for "windows or osx". Looks like we use `defined(__APPLE__)` elsewhere so that should be fine for OSX.

    #if defined(WIN32) || defined(__APPLE__)